File: Lemonade Change - GFG/lemonade-change.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
bool lemonadeChange(int N, vector<int> &bills) {
        int five=0;
        int ten=0;
        int twenty=0;
        for(int i=0;i<bills.size();i++){
            if(bills[i]==5){
                five++;
            }
            if(bills[i]==10){
                if(five==0) return false;
                ten++;
                five--;
            }
            if(bills[i]==20){
                if(ten==0){
                    if(five<3) return false;
                    else{
                        five=five-3;
                    }
                }
                if(ten>0){
                    if(five==0)return false;
                    if(five>0){
                        ten--;
                        five--;
                    }
                }
            }
        }
        return true;
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int N;
        cin >> N;

        vector<int> bills(N);
        for (int i = 0; i < N; i++) cin >> bills[i];

        Solution obj;
        int ans = obj.lemonadeChange(N, bills);
        if (ans)
            cout << "True" << endl;
        else
            cout << "False" << endl;
    }
    return 0;
}
// } Driver Code Ends

File: Lemonade Change - GFG/lemonade-change.py
#{ 
 # Driver Code Starts
#Initial Template for Python 3

# } Driver Code Ends
#User function Template for python3
class Solution:
    def lemonadeChange(self, N, bills):
        count_5 = 0
        count_10 = 0
        count_20 = 0
        for bill in bills:
            if bill == 5:
                count_5 += 1
            elif bill == 10:
                if count_5 >= 1:
                    count_5 -= 1
                    count_10 += 1
                else:
                    return False
            elif bill == 20:
                if count_10 >= 1 and count_5 >= 1:
                    count_10 -= 1
                    count_5 -= 1
                    count_20 += 1
                elif count_5 >= 3:
                    count_5 -= 3
                    count_20 += 1
                else:
                    return False
        return True

        
#{ 
 # Driver Code Starts.
if __name__ == '__main__': 
    t = int(input ())
    for _ in range (t):
        N = int(input())
        bills = list(map(int, input().split()))
        ob = Solution()
        res = ob.lemonadeChange(N, bills)
        print(res)
# } Driver Code Ends